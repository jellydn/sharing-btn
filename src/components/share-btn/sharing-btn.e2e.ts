import { newE2EPage } from '@stencil/core/testing';

describe('sharing-btn', () => {
  it('renders', async () => {
    const page = await newE2EPage();

    await page.setContent('<sharing-btn></sharing-btn>');
    const element = await page.find('sharing-btn');
    expect(element).toHaveClass('hydrated');
  });

  it('renders changes to the url and description', async () => {
    const page = await newE2EPage();

    await page.setContent('<sharing-btn></sharing-btn>');
    const component = await page.find('sharing-btn');
    const element = await page.find('sharing-btn >>> div');
    expect(element.textContent).toMatchSnapshot();

    component.setProperty('url', 'https://stenciljs.com');
    await page.waitForChanges();
    expect(element.textContent).toMatchSnapshot();

    component.setProperty('description', "Hello, World! I'm James Quincy");
    await page.waitForChanges();
    expect(element.textContent).toMatchSnapshot();
  });
});
