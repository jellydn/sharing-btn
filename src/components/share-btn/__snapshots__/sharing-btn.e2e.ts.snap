// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sharing-btn renders changes to the url and description 1`] = `"FacebookTwitterE-Mail"`;

exports[`sharing-btn renders changes to the url and description 2`] = `"FacebookTwitterE-Mail"`;

exports[`sharing-btn renders changes to the url and description 3`] = `"FacebookTwitterE-Mail"`;
